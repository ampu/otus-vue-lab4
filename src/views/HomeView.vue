<template>
  <main class="h-100 d-flex flex-column justify-content-center align-items-center mb-5">
    <div class="d-flex flex-column justify-content-center align-items-center mb-5">
      <h1 class="m-5">Bookstore «Gallery»</h1>
      <ul class="links d-flex align-items-center gap-5">
        <li
          v-for="{name, title, icon} of LINKS"
          :key="name"
        >
          <RouterLink
            :data-testid="`home-${name}`"
            :to="{name}"
            class="link nav-link"
          >
            <component :is="icon"/>
            {{ title }}
          </RouterLink>
        </li>
      </ul>
    </div>
  </main>
</template>

<script setup lang="ts">
import BooksIcon from '@/assets/icons/books.svg'
import AuthorsIcon from '@/assets/icons/authors.svg'

const LINKS = [
  {name: `books`, title: `Books`, icon: BooksIcon},
  {name: `authors`, title: `Authors`, icon: AuthorsIcon},
]
</script>

<style lang="scss" scoped>
.links {
  list-style: none;
  margin: 0;
  padding: 0;
}

.link {
  display: flex;
  align-items: center;
  gap: 5px;

  font-size: 2em;
  text-decoration: none;

  opacity: 0.8;
  transition: opacity 0.3s;

  &:hover {
    opacity: 1;
  }
}
</style>
