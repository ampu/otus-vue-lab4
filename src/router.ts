import {createRouter, createWebHashHistory} from 'vue-router'

import HomeView from '@/views/HomeView.vue'
import NotFoundView from '@/views/NotFoundView.vue'

const routes = [
  {
    path: '/:pathMatch(.*)*',
    name: `not-found`,
    component: NotFoundView,
  },
  {
    path: `/`,
    name: `home`,
    component: HomeView
  },
  // books
  {
    path: `/books`,
    name: `books`,
    component: () => import(/* webpackChunkName: "books" */ '@/views/BooksView.vue')
  },
  {
    path: `/books/add`,
    name: `books-add`,
    component: () => import(/* webpackChunkName: "books-add" */ '@/views/AddBookView.vue')
  },
  {
    path: `/books/:bookId/edit`,
    name: `books-edit`,
    component: () => import(/* webpackChunkName: "books-edit" */ '@/views/EditBookView.vue'),
    props: true,
  },
  // authors
  {
    path: `/authors`,
    name: `authors`,
    component: () => import(/* webpackChunkName: "authors" */ '@/views/AuthorsView.vue')
  },
  {
    path: `/authors/add`,
    name: `authors-add`,
    component: () => import(/* webpackChunkName: "authors-add" */ '@/views/AddAuthorView.vue')
  },
  {
    path: `/authors/:authorId/edit`,
    name: `authors-edit`,
    component: () => import(/* webpackChunkName: "authors-edit" */ '@/views/EditAuthorView.vue'),
    props: true,
  },
]

const router = createRouter({
  history: createWebHashHistory(import.meta.env.BASE_URL),
  routes,
  scrollBehavior(to, from, savedPosition) {
    return savedPosition || {left: 0, top: 0, behavior: `smooth`}
  },
})

Object.assign(window, {router})

export default router
