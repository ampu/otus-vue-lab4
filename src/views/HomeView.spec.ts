import {describe, it, expect} from 'vitest'
import {shallowMount} from '@vue/test-utils'

import HomeView from '@/views/HomeView.vue'
import router from '@/router'

describe(`HomeView`, () => {
  const mountComponent = () => {
    return shallowMount(HomeView, {
      global: {
        plugins: [router],
      },
    })
  }

  it(`imports without error`, () => {
    expect(HomeView).toBeTypeOf(`object`)
  })

  it(`mounts without error`, () => {
    const wrapper = mountComponent()
    expect(wrapper).toBeDefined()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
