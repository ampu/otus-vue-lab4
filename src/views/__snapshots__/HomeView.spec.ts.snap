// Vitest Snapshot v1

exports[`HomeView > mounts without error 1`] = `
"<main data-v-c3e9f06f=\\"\\" class=\\"h-100 d-flex flex-column justify-content-center align-items-center mb-5\\">
  <div data-v-c3e9f06f=\\"\\" class=\\"d-flex flex-column justify-content-center align-items-center mb-5\\">
    <h1 data-v-c3e9f06f=\\"\\" class=\\"m-5\\">Bookstore «Gallery»</h1>
    <ul data-v-c3e9f06f=\\"\\" class=\\"links d-flex align-items-center gap-5\\">
      <li data-v-c3e9f06f=\\"\\">
        <router-link-stub data-v-c3e9f06f=\\"\\" to=\\"[object Object]\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\" data-testid=\\"home-books\\" class=\\"link nav-link\\"></router-link-stub>
      </li>
      <li data-v-c3e9f06f=\\"\\">
        <router-link-stub data-v-c3e9f06f=\\"\\" to=\\"[object Object]\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\" data-testid=\\"home-authors\\" class=\\"link nav-link\\"></router-link-stub>
      </li>
    </ul>
  </div>
</main>"
`;
